salt/utils/data.py

```diff
@@ -286,6 +287,8 @@ def decode(
                 to_str,
             )
         )
+    if isinstance(data, datetime.datetime):
+        return str(data)
```

Comment:
Should we instead call [`.isoformat()`](https://docs.python.org/3/library/datetime.html#datetime.datetime.isoformat)?

The added value is that it can be [parsed back to a datetime object](https://docs.python.org/3/library/datetime.html#datetime.datetime.fromisoformat) if needed.

Reply:
@s0undt3ch That works better.  Will update.